vm.go

```diff
@@ -227,6 +228,19 @@ type instruction interface {
 	exec(*vm)
 }
 
+func (vm *vm) getFuncName() unistring.String {
+	vm.funcNameLock.Lock()
```

Comment:
It looks like we're using the write lock here and on line 234. You'll want this to be: 
```suggestion
	vm.funcNameLock.RLock()
```

Reply:
🤦 thanks for catching that!